### DISCUSSION TITLE
whos enhancement to pyttb utils

### DISCUSSION THREAD

**Discussion by DeepBlockDeepak** [sandialabs/pyttb#136] [Ideas]
Though there isn't a direct Python analogue to [MATLAB's `whos`](https://www.mathworks.com/help/matlab/ref/whos.html), this is a first stab at how to deal with it in Python. This specifically regards the `whos` use in the [ttb docs](https://tensortoolbox.org/tensor_doc.html#6). 

At the moment it looks like:
```python
def whos(*args):
    print(f"{'Name': <10}{'Size': <10}{'Bytes': <10}{'Class': <10}{'Num. Attributes': <5}")
    print("-"*55)
    for name in args:
        var = globals()[name]
        size = 'x'.join(map(str, var.shape)) if hasattr(var, 'shape') else 'N/A'
        attributes = [attr for attr in dir(var) if not attr.startswith('__')]
        print(f"{name: <10}{size: <10}{str(sys.getsizeof(var)): <10}{str(type(var).__name__): <10}{len(attributes): <5}")

```
* `whos()` can take any number of arguments, like in the ttb doc where two tensor class objects are passed.
* The global symbol table allows us to fetch the corresponding variables in `args`.
* The size of each variable is obtained, so long as it has the tensor-class' `shape` attribute.  Since `shape` is a tuple, its elements are mapped to string and then joined to give us the MATLAB-styled tensor dimensions. 
* Since I couldn't seem to find a direct analogue to the MATLAB-whos [Attribute](https://www.mathworks.com/help/matlab/ref/whos.html#bu2_rvt) category, I decided to count the number of attributes belonging to the object, filtering out the dunder methods.
* The print statement aligns the data to the column names. The `Class` field uses only the name of the type class for brevity.

**Comment by dmdunla** [comment]
The `%whos` Jupyter notebook command provides sufficient functionality for now:

```
import pyttb as ttb
X = ttb.tenrand((2,2,2))
```

```
%whos

ariable   Type      Data/Info
------------------------------
X          tensor    tensor of shape 2 x 2 x 2<...>0.39772558 0.41173942]]\n
ttb        module    <module 'pyttb' from 'C:\<...>ttb\\pyttb\\__init__.py'>
```

**Comment by dmdunla** [comment]
See PR #135 for an draft implementation.
